Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -1074,6 +1083,41 @@ void HackStudioWidget::initialize_menubar(GUI::Menubar& menubar)
     create_help_menubar(menubar);
 }
 
+void HackStudioWidget::handle_external_file_deletion(const String& filepath)
+{
+    struct stat st;
+    if (lstat(filepath.characters(), &st) == 0) {
+        return;
+    }
```

Comment:
Let's do the `lstat` check before calling `handle_external_file_deletion`.
Currently `handle_external_file_deletion` gets called whenever a file changes.

Reply:
Also, let's explicitly check if `errno` is `ENOENT` to determine that the file was deleted.